syncmaster/backend/__init__.py

```diff
@@ -36,22 +37,26 @@ def application_factory(settings: Settings) -> FastAPI:
         allow_headers=["*"],
     )
 
+    application.add_middleware(
+        CorrelationIdMiddleware,
+        header_name="X-Correlation-ID",
+        generator=lambda: str(uuid.uuid4()),
+    )
+
     application.include_router(api_router)
     application.exception_handler(RequestValidationError)(validation_exception_handler)
     application.exception_handler(ValidationError)(validation_exception_handler)
     application.exception_handler(SyncmasterError)(syncmsater_exception_handler)
     application.exception_handler(HTTPException)(http_exception_handler)
     application.exception_handler(Exception)(unknown_exception_handler)
```

Comment:
My recommendation is to move these middlewares into separated file, and just call registration function here:
- https://github.com/MobileTeleSystems/data-rentgen/tree/4e615d1f14d21470a41755c394d1d9c149b9b7dd/data_rentgen/server/middlewares
- - https://github.com/MobileTeleSystems/data-rentgen/blob/4e615d1f14d21470a41755c394d1d9c149b9b7dd/data_rentgen/server/__init__.py#L31-L32

Reply:
as middleware configuration has many intersections with settings, I decided to put them in current PR to escape from resolving conflicts in future